src/Angor/Client/Pages/Signatures.razor

```diff
@@ -357,26 +440,117 @@
                 sigJson);
 
             FounderProject.LastRequestForSignaturesTime = SignService.SendSignaturesToInvestor(encryptedContent, nostrPrivateKeyHex, signature.investorPubKey, signature.EventId);
-
             Storage.UpdateFounderProject(FounderProject);
 
             signaturesRequests.Single(_ => _.investorPubKey == signature.investorPubKey && _.TimeApproved is null)
                 .TimeApproved = FounderProject.LastRequestForSignaturesTime;
 
             return new OperationResult { Success = true };
-        });
+        }
+        catch (Exception ex)
+        {
+            return new OperationResult { Success = false, Message = $"An error occurred: {ex.Message}" };
+        }
+    }
+
+
+
+
+
 
-        if (operationResult.Success)
+    private async Task ApproveSignature(SignatureRequest signature)
+    {
+        var words = await passwordComponent.GetWalletAsync();
+        if (words == null)
+        {
+            Logger.LogDebug("words is null.");
+            passwordComponent.ShowPassword(async () =>
+            {
+                await ApproveSignature(signature);
+            });
+            return;
+        }
+
+        var operationResult = await PerformSignatureApproval(signature, words);
+
+        if (!operationResult.Success)
         {
+            notificationComponent.ShowErrorMessage(operationResult.Message);
+        }
+        else
+        {
+            StateHasChanged();
+        }
+    }
+
+
+
+
+
+    private async Task ApproveAllSignatures()
+    {
+        isLoading = true;
+        StateHasChanged();
+
+        try
+        {
+            if (passwordComponent.HasPassword())
+            {
+                await ProcessSignatures();
+            }
+            else
+            {
+                passwordComponent.ShowPassword(async () =>
+                {
+                    await ProcessSignatures();
+                });
+            }
+
             passwordComponent.ClearPassword();
+        }
+        catch (Exception e)
+        {
+            Logger.LogDebug($"Error approving signatures: {e.Message}");
+        }
+        finally
+        {
+            isLoading = false;
             StateHasChanged();
         }
-        else
+    }
+
+
+
+    private async Task ProcessSignatures()
+    {
+        var pendingSignatures = signaturesRequests.Where(s => s.TransactionHex != null && s.AmountToInvest != null && s.TimeApproved == null).ToList();
+        numOfSignatureToSign = pendingSignatures.Count;
+        numOfSignaturesSigned = 0;
+        
+        var words = await passwordComponent.GetWalletAsync();
+        if (words == null)
```

Comment:
this if can be deleted as a whole as it is a fail safe, depends how we want to treat this.